spec.bs

```diff
@@ -4004,64 +4041,96 @@ from querying the server during an auction.
+      * The result of [=compute the key part for one of multiple reporting ids=] given |igAd|'s 
+        [=interest group ad/buyer and seller reporting ID=]
+      * The result of [=compute the key part for one of multiple reporting ids=] given |igAd|'s
+        [=interest group ad/buyer reporting ID=]
+    1. Otherwise: 
+      1. If |igAd|'s [=interest group ad/buyer and seller reporting ID=] is not null, set |keyString|
+        to be the [=string/concatenation=] of the following strings separated with U+000A (LF):
+        * "BuyerAndSellerReportId"
+        * |middle|
+        * |igAd|'s [=interest group ad/buyer and seller reporting ID=]
+      1. Otherwise if |igAd|'s [=interest group ad/buyer reporting ID=] is not null, set |keyString|
+        to be the [=string/concatenation=] of the following strings separated with U+000A (LF):
+        * "BuyerReportId"
+        * |middle|
+        * |igAd|'s [=interest group ad/buyer reporting ID=]
+      1. Otherwise set |keyString| to be the [=string/concatenation=] of the following strings separated
+        with U+000A (LF):
+        * "NameReport"
+        * |middle|
+        * |igAd|'s [=interest group/name=]
+    1. Return the [=SHA-256=] hash of the [=ASCII encoding=] of |keyString|.
 </div>
 
 <div algorithm>
  To <dfn>compute the key hash of component ad</dfn> given an [=interest group ad=] |igAd|:
    1. Let |keyString| be the [=string/concatenation=] of the following strings separated with U+000A LF:
-      1. "ComponentBid"
-      1. the [=URL serializer|serialization=] of |igAd|.
+      * "ComponentBid"
+      * the [=URL serializer|serialization=] of |igAd|.
    1. Return the [=SHA-256=] hash of the [=ASCII encoding=] of |keyString|.
 </div>
 
 <div algorithm>
   To <dfn>query generated bid k-anonymity count</dfn> given a [=generated bid=] |bid|:
-    1. Compute the |adHashCode| following [=compute the key hash of ad=] with the |bid|'s [=generated bid/interest group=] and |bid|'s [=generated bid/ad descriptor=].
+    1. Compute the |adHashCode| following [=compute the key hash of ad=] with the |bid|'s [=generated bid/interest group=], 
+      |bid|'s [=generated bid/ad descriptor=].
     1. If [=query k-anonymity cache=] for |adHashCode| returns false, return false.
     1. If |bid|'s [=generated bid/ad component descriptors=] is not null:
       1. [=set/For each=] |adComponentDescriptor| in |bid|'s
         [=generated bid/ad component descriptors=]:
         1. Compute the |componentAdHashCode| by getting the result of [=compute the key hash of component ad=] with |adComponentDescriptor|'s
           [=ad descriptor/url=].
         1. If [=query k-anonymity cache=] for |componentAdHashCode| returns false, return false.
+    1. If |bid|'s [=generated bid/selected buyer and seller reporting ID=] is not null:
+      1. Let |reportingHashCode| be the result of [=compute the key hash of reporting ID=] with the |bid|'s [=generated bid/interest group=],

```

Comment:
This is missing the ad argument (the second argument). Also, shouldn't this use `[=query reporting ID k-anonymity count=]`?